skiplist/skip_list_map.go

```diff
@@ -69,69 +72,71 @@ func (it *SkipListIterator) Next() (interface{}, interface{}, error) {
 	return cur.key, cur.value, nil
 }
 
-type SkipListMapI interface {
+type SkipListMapI[K comparable, V any] interface {
 	Size() int
 
 	// Insert key/value into the list.
 	// REQUIRES: nothing that compares equal to key is currently in the list.
-	Insert(key interface{}, value interface{})
+	Insert(key K, value V)
 
-	// Returns true if an entry that compares equal to key is in the list.
-	Contains(key interface{}) bool
+	// Contains Returns true if an entry that compares equal to key is in the list.
+	Contains(key K) bool
 
-	// Returns the value element that compares equal to the key supplied or returns NotFound if it does not exist.
-	Get(key interface{}) (interface{}, error)
+	// Get Returns the value element that compares equal to the key supplied or returns NotFound if it does not exist.
+	Get(key K) (V, error)
```

Comment:
TODO take a look at how the btree does it like here: https://github.com/etcd-io/etcd/pull/14515/files#diff-540e25989cc5e6d942a6c05ad0fc9d6ee3ae26fc8d0ccb14c3919f5b7b3c76deR59

Reply:
https://github.com/google/btree/pull/45